internal/service/federatedsettingsorgconfig/resource_federated_settings_connected_org.go

```diff
@@ -119,27 +139,38 @@ func resourceMongoDBAtlasFederatedSettingsOrganizationConfigUpdate(ctx context.C
 		domainAllowList := d.Get("domain_allow_list")
 		federatedSettingsConnectedOrganizationUpdate.SetDomainAllowList(cast.ToStringSlice(domainAllowList))
 	}
+	if d.HasChange("data_access_identity_provider_ids") {
+		dataAccessIdentityProviderIDs := d.Get("data_access_identity_provider_ids")
+		federatedSettingsConnectedOrganizationUpdate.SetDataAccessIdentityProviderIds(cast.ToStringSlice(dataAccessIdentityProviderIDs))
+	}
 
 	if d.HasChange("identity_provider_id") {
 		identityProviderID := d.Get("identity_provider_id").(string)
-		federatedSettingsConnectedOrganizationUpdate.SetIdentityProviderId(identityProviderID)
+		// if identityProviderId is not part of the PATCH payload, it will be detached, "" will raise VALIDATION_ERROR
+		if identityProviderID == "" {
+			federatedSettingsConnectedOrganizationUpdate.IdentityProviderId = nil
```

Comment:
@oarbusi and me were discussing this. But we thought it is better to leave it as `required` to avoid changes and since an empty value will detach the `identity_provider_id` see last `note` in [API DOCS](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Federated-Authentication/operation/updateConnectedOrgConfig)

Reply:
See also the third note in the api docs:

> If the identityProviderId field is not provided, you will disconnect the organization and the identity provider.

Would disconnecting the organization and the identity provider be a valid use case? If it is, supporting this by having to leave an empty string sounds strange. To keep in mind passing from required to optional would not impact existing configurations.

Reply:
I agree, having to leave an empty string is a strange ux.
Updated to `Optional` in b959c8b